x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/DataNodeRequestSender.java

```diff
@@ -265,9 +265,8 @@ void onAfter(DriverCompletionInfo info) {
                     concurrentRequests.release();
                 }
 
-                if (pendingRetries.isEmpty() == false && remainingUnavailableShardResolutionAttempts.decrementAndGet() >= 0) {
+                if (sendingLock.isHeldByCurrentThread()) {
```

Comment:
I think `isHeldByCurrentThread` should be used for assertions or debugging purposes, not in production code.

Reply:
Agree. let me find a way to replace it.